### ISSUE TITLE
Dont require escaping of int/float

### ISSUE THREAD

**Issue by kkmuffme** [WordPress/WordPress-Coding-Standards#1888]
## Bug Description
int/float type does not need to be escaped before output
WordPress does not even provide an escaping function for this.

Using esc_html/esc_attr is incorrect, since their param type is string only.

## Minimal Code Snippet
```php
$i = 3;
echo $i;
```

## Error Code
WordPress.Security.EscapeOutput.OutputNotEscaped

**Comment by dingo-d** [comment]
Why would you write your code like that? I mean this example alone is not enough to warrant the change, as it can be written as just `echo 3;`. 

Can you provide some other, more real-life, example?

**Comment by kkmuffme** [comment]
Because this is the minimum to reproduce the issue.
This issue affects any place that echos int/floats.

If you want a more complex example, here's a simple polling/queuing example:

```php
/**
 * Increment a counter by the amount specified
 *
 * @param string|int cache key
 * @param int        $offset amount by which we should decrease/increase the stored value
 * @param string     $group the cache key group used to store the value
 * @return int|bool
 */
function increment( $key, $offset = 1, $group ) {
    $value = get( $key . $group );
    $value += $offset;
    $value = apply_filters( 'some_action_for_incr', $value );
    return $value;
}

function my_demo( $value ) {
    if ( $key == 'blabla' && $value > 500 ) {
        // in reality this sets of an async task, which in terms will then echo the $value. but you get the issue like this too, so I kept it simple.
        trigger_error( 'blabla is ' . $value, E_USER_WARNING );
    }
    return $value;
}
add_filter( 'some_action_for_incr', 'my_demo' );
```

**Comment by jrfnl** [comment]
The problem is that a variable can be defined anywhere and, especially if it's being filtered, there is no guarantee that the type of the variable will still be float by the time the variable is output.

As this is a static analysis tool, the runtime value of variables is not available, so this is impossible to verify.

The simple way around it, though, is to just cast to float/int instead of using any of the output escaping functions. Casts to float/int will be accepted as valid output and the variable will then be regarded as safe by the sniff.

```php
        trigger_error( 'blabla is ' . (int) $value, E_USER_WARNING );
```

**Comment by kkmuffme** [comment]
Ok thx for the suggestion.

But for simple things that stay within a functions context phpcs couldn't check for it?
e.g. the most rudimentary checking case is something like:

```php
$counter = 0;
while( true ) {
    if ( in_sync() ) {
          break;
    } elseif ( $counter > 30 ) {
          trigger_error( 'sync took longer than ' . $counter, E_USER_WARNING );
          break;
    }
    sleep( 1 );
    $counter++;
}
```

**Comment by jrfnl** [comment]
While your example would be checkable, in nearly all cases, it will be impossible to check the type of a variable.

To try and trace the variable type within a function is hard enough from within a sniff, doing so when it won't be possible to determine this reliably in 95% of all cases, is an exercise in futility.

```php
$counter = 0;
while( true ) {
    if ( in_sync() ) {
          $counter = false; // As the sniff can't determine whether the `if` condition will be `true`, it can't determine whether the variable type is changed or not.
          break;
    } elseif ( $counter > 30 ) {
          $counter = apply_filters( 'counter_filter', $counter ); // The sniff won't be able to determine whether the variable type has changed or not.
          trigger_error( 'sync took longer than ' . $counter, E_USER_WARNING );
          break;
    }
    sleep( 1 );
    $counter++;
}
```

**Comment by kkmuffme** [comment]
Ok, it's just bc I also use psalm/phpstan and both keep track of the int, so when I use the esc_html than psalm/phpstan compains (rightly so) that esc_html doesn't accept int, but when I don't use esc_html then phpcs complains...
So I guess only solution is to typecast to (int)
